File: src/main/java/org/antop/java/chapter2/BinaryReps.java
package org.antop.java.chapter2;

import java.util.Map;
import java.util.Set;
import java.util.TreeMap;

public class BinaryReps {

    public static void main(String[] args) {
        var binaryReps = new TreeMap<Character, String>();
        for (char c = 'A'; c <= 'F'; c++) {
            var binary = Integer.toBinaryString(c);
            binaryReps.put(c, binary);
        }
        for (Map.Entry<Character, String> e : binaryReps.entrySet()) {
            var letter = e.getKey();
            var binary = e.getValue();
            System.out.println(letter + " = " + binary);

            var number = 10;
            var percentage = number;
            if (percentage < 0 && percentage > 100) {
                throw new IllegalArgumentException("" + number);
            }
        }
    }

}

File: src/main/kotlin/org/antop/kotlin/chapter2/BinaryReps.kt
package org.antop.kotlin.chapter2

import java.util.*

fun main() {
    val binaryReps = TreeMap<Char, String>()
    for (c in 'A'..'F') {
        val binary = Integer.toBinaryString(c.code)
        binaryReps[c] = binary
    }
    for ((letter, binary) in binaryReps) {
        println("$letter = $binary")
    }
    // 사용하지 않는 변수는 밑줄로 처리
    for ((_, binary) in binaryReps) {
        println(binary)
    }
}
